glue_runner.py

```diff
@@ -0,0 +1,276 @@
+            # set working directory to this script's directory
+            cwd=f"{this_script_dir}",
+        )
+        while True:
+            line = proc.stdout.readline().decode("utf-8")
+            print(f"{line}")
+            if not line and proc.poll() is not None:
+                break
+            line = line.strip()
+            if line.find("DELTACAT_PRESIGNED_URL=") != -1:
+                dc_url = line.split("=", 1)[1]
+        returncode = proc.poll()
+        if returncode != 0:
+            raise RuntimeError(f"Command {cmd} Failed. Exit Code: {returncode}")
+
+    s3_client = boto3.client("s3")
+
+    # upload the example runner script to S3
+    bucket_name = f"deltacat-glue-scripts-{stage}"
+    print(f"Ensuring Glue Job Script Bucket exists: {bucket_name}")
+    s3_client.create_bucket(Bucket=bucket_name)
+    print(f"Using Glue Job Script Bucket: {bucket_name}")
+
+    # create or update the glue job
+    glue_job_name = args.glue_job_name
+    if not glue_job_name:
+        glue_job_name = f"deltacat-example-runner-{stage}"
+    run_script_path = os.path.relpath(args.script_path)
+    s3_glue_script_file_path = f"s3://{bucket_name}/{run_script_path}"
+    print(f"Uploading script '{run_script_path}' to '{s3_glue_script_file_path}'")
+    s3_client.upload_file(
+        run_script_path,
+        bucket_name,
+        run_script_path,
+    )
+    print(f"Uploaded Glue Job Script to: {s3_glue_script_file_path}")
+    glue_max_workers = args.glue_max_workers
+    glue_iam_role = args.glue_iam_role
+    glue_job_config = {
+        "Role": glue_iam_role,
+        "GlueVersion": "4.0",
+        "ExecutionProperty": {"MaxConcurrentRuns": 10},
+        "WorkerType": "Z.2X",
+        "NumberOfWorkers": glue_max_workers,
+        "Command": {
+            "Name": "glueray",
+            "Runtime": "Ray2.4",
+            "PythonVersion": "3.9",
+            "ScriptLocation": s3_glue_script_file_path,
+        },
+    }
+    region = args.region
+    glue_client = boto3.client(
+        "glue",
+        region,
+    )
+    try:
+        response = glue_client.get_job(JobName=glue_job_name)
+        print(f"Found Existing Glue Job `{glue_job_name}`: {response}")
+        print(f"Updating Existing Glue Job `{glue_job_name}`")
+        if not glue_iam_role:
+            glue_job_config["Role"] = response["Job"]["Role"]
+        glue_client.update_job(
+            JobName=glue_job_name,
+            JobUpdate=glue_job_config,
+        )
+    except ClientError as e:
+        print(f"Glue Job `{glue_job_name}` doesn't exist. Creating it.")
+        # create or update a glue job
+        glue_client.create_job(
+            Name=glue_job_name,
+            **glue_job_config,
+        )
+    print(f"New Glue Job Config: {glue_job_config}")
+
+    print(f"Getting Last Job Run Arguments for {glue_job_name}")
+    response = glue_client.get_job_runs(
+        JobName=glue_job_name,
+    )
+    job_runs = response["JobRuns"]
```

Comment:
would validate response for `JobRuns` key before trying to access. It could be quite possible the first iteration of a new Job does not have any Job runs.

Reply:
Good catch - will fix this to remove the assumption that job runs exist.